plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs

```diff
@@ -27,12 +27,13 @@ betaStep = \case
             bindings = binding :| []
         in
             Let a NonRec bindings body
-    TyInst a (TyAbs _ tyname kind body) typ ->
-        let tyVarDecl = TyVarDecl a tyname kind
-            tyBinding = TypeBind a tyVarDecl typ
-            bindings  = tyBinding :| []
-        in
-            Let a NonRec bindings body
+    -- This case is disabled as it introduces a lot of type inlining (determined from profiling)
```

Comment:
Rather link to https://input-output.atlassian.net/browse/SCP-2570, the transformation is currently unsound! (will turn typechecking programs into non-typechecking programs)

Reply:
Yeah, it makes a dramatic improvement reducing the compilation time of the marlowe contract from 15s to 2s.

Reply:
So there are a few problems here that this is covering up:
- This transformation is unsound
- The inlining for types is too eager and needs some heuristics, so overall we were inlining way too much
- We probably do too much renaming when we inline

This change happens to stop all that, but I want to tackle all of those issues separately. But even if we do all that I think we'll end up doing quite a lot of renaming, so I think we might as well make it faster.